waspc/test/Generator/CrudTest.hs

```diff
@@ -0,0 +1,232 @@
+        crudOperationsName
+        AS.Crud.Crud
+          { entity = AS.Core.Ref.Ref crudOperationEntitName,
+            operations =
+              AS.Crud.CrudOperations
+                { get =
+                    Just
+                      ( AS.Crud.CrudOperationOptions
+                          { isPublic = Just True,
+                            overrideFn = Nothing
+                          }
+                      ),
+                  getAll =
+                    Just
+                      ( AS.Crud.CrudOperationOptions
+                          { isPublic = Just False,
+                            overrideFn = Nothing
+                          }
+                      ),
+                  create =
+                    Just
+                      ( AS.Crud.CrudOperationOptions
+                          { isPublic = Just True,
+                            overrideFn = Nothing
+                          }
+                      ),
+                  update = Nothing,
+                  delete = Nothing
+                }
+          }
+        primaryEntityField
+        `shouldBe` mkOperationsJson
+          ( object
+              [ "Get"
+                  .= object
+                    [ "route" .= ("get" :: String),
+                      "fullPath" .= ("tasks/get" :: String),
+                      "isPublic" .= True
+                    ],
+                "GetAll"
+                  .= object
+                    [ "route" .= ("get-all" :: String),
+                      "fullPath" .= ("tasks/get-all" :: String),
+                      "isPublic" .= False
+                    ],
+                "Create"
+                  .= object
+                    [ "route" .= ("create" :: String),
+                      "fullPath" .= ("tasks/create" :: String),
+                      "isPublic" .= True
+                    ]
+              ]
+          )
+
+    it "the override import can defined" $ do
+      getCrudOperationJson
+        crudOperationsName
+        AS.Crud.Crud
+          { entity = AS.Core.Ref.Ref crudOperationEntitName,
+            operations =
+              AS.Crud.CrudOperations
+                { get =
+                    Just
+                      ( AS.Crud.CrudOperationOptions
+                          { isPublic = Just True,
+                            overrideFn =
+                              Just $
+                                AS.ExtImport.ExtImport
+                                  { AS.ExtImport.name = AS.ExtImport.ExtImportField "getTask",
+                                    AS.ExtImport.path = SP.castRel [relfileP|bla/tasks.js|]
+                                  }
+                          }
+                      ),
+                  getAll =
+                    Just
+                      ( AS.Crud.CrudOperationOptions
+                          { isPublic = Just False,
+                            overrideFn = Nothing
+                          }
+                      ),
```

Comment:
You have a lot of these expressive structures, which of course makes sense, you are describing AppSpec, but it might make sense to introduce some helper functions to reduce the amount of code here? `operation (Just False) Nothing` or smth like that? Maybe not, maybe that is not extensible here, but wanted to possibly give an idea.
Oh, or maybe you can define default `crud operation option and` and then just add to it with `{ somestuff = something }`. Oh yeah that is probably better. SO basically you would have `defaultOperation = { isPublic = Nothing, overrideFn = Nothing }` and then you can do `defaultOperation { isPublic = Just False }`. You could even go all the way with something like `defop { isPublic = Just False }`.